test/libyul/yulOptimizerTests/expressionSimplifier/inside_for.yul

```diff
@@ -1,12 +1,13 @@
 {
-    for { let a := 10 } iszero(eq(a, 0)) { a := add(a, 1) } {}
+    let a := 10
```

Comment:
I am not very familiar with the ticket you're solving, but I believe that this should still work after the changes. @ekpyron confirm please? (ticket in question is #6202)

Reply:
Ah, but because that is done in a different optimization step it won't work in the individual optimization tests of other steps.. that could make sense..

Reply:
Well, it depends :-). The ticket is meant to remove special treatment of pre-blocks in all optimizer components and rather let them depend on the fact that pre blocks were already removed before in the ``ForLoopInitRewriter``.
The tests that only call one isolated optimizer component need to deal with that dependency - so either we don't use pre-blocks in such tests (that is this change here) or we call the ``ForLoopInitRewriter`` before running such a test run in ``YulOptimizerTest.cpp``. Both options are fine and I wouldn't prefer either, so we can leave it at this.

Reply:
In short: It's fine as it is ;-).